scripts/profiler/model_creator.py

```diff
@@ -0,0 +1,296 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+import time
+import sys
+from enum import Enum
+import logging
+import json
+import socket
+from rtt_nordic_config import RttNordicConfig
+from events import Event, EventType, TrackedEvent
+from processed_events import ProcessedEvents, MEM_ADDRESS_DATA_DESC
+from io import StringIO
+import csv
+import errno
+
+RECV_BUF_SIZE = 2**13
+
+class Command(Enum):
+    START = 1
+    STOP = 2
+    INFO = 3
+
+class ModelCreator:
+
+    def __init__(self, sock_addr_dict,
+                 config=RttNordicConfig,
+                 event_filename=None,
+                 event_types_filename=None,
+                 log_lvl=logging.INFO):
+        self.config = config
+        self.event_filename = event_filename
+        self.event_types_filename = event_types_filename
+        self.csvfile = None
+
+        self.sock_addr_dict = sock_addr_dict
+        self.sock_desc = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
```

Comment:
We use similar approach related to forwarding data between various Python processes.
Maybe we could provide some abstraction related to this operation (e.g. providing a class for "data and description stream")? The API could provide functions to get/put event and get/put event description using socket. The socket descriptions could be configured during initialization as class instance specific values.
That could make data transferring between Python modules more generic (so it would be easier to integrate new modules), data and its description would be bundled together too.